pkg/rbac/parser.go

```diff
@@ -44,8 +48,62 @@ type Rule struct {
 	URLs      []string `marker:"urls,optional"`
 }
 
+// target generates a string representing the resources and/or non-resources a Rule applies.
+func (r *Rule) target() string {
+	r.cleanUp()
+	s := []string{
+		strings.Join(r.Groups, ","),
+		strings.Join(r.Resources, ","),
+		strings.Join(r.URLs, ","),
+	}
+	return strings.Join(s, ",")
+}
+
+// addVerbs adds new verbs into a Rule
+func (r *Rule) addVerbs(verbs []string) {
+	r.Verbs = removeDup(append(r.Verbs, verbs...))
+}
+
+// cleanUp removes duplicates from each field of a Rule, and sorts each field.
+func (r *Rule) cleanUp() {
+	r.Groups = removeDup(r.Groups)
+	r.Resources = removeDup(r.Resources)
+	r.Verbs = removeDup(r.Verbs)
+	r.URLs = removeDup(r.URLs)
+	sort.Strings(r.Groups)
+	sort.Strings(r.Resources)
+	sort.Strings(r.Verbs)
+	sort.Strings(r.URLs)
+}
+
+// merge merges another Rule into the existing Rule
+// Only the Rules with the same Rule.target() can be merged.
+func (r *Rule) merge(other Rule) error {
+	if r.target() != other.target() {
```

Comment:
why does this need an extra check?  Maybe just get rid of this method.

Reply:
Removed this method.